File: src/main/java/um/tesoreria/core/hexagonal/ubicacion/domain/model/Ubicacion.java
package um.tesoreria.core.hexagonal.ubicacion.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Ubicacion {
    private Integer ubicacionId;
    private String nombre;
    private Integer dependenciaId;
    private Integer geograficaId; // Derived from Dependencia if needed
}

File: src/main/java/um/tesoreria/rest/kotlin/model/Ubicacion.kt
package um.tesoreria.rest.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table

@Entity
@Table
data class Ubicacion(

    @Id
    var ubicacionId: Int? = null,

    var nombre: String = "",
    var dependenciaId: Int? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null

) : Auditable()
